src/ADNLPProblems/camshape.jl

```diff
@@ -0,0 +1,42 @@
+export camshape
+
+function camshape(args...; n::Int = default_nvar, type::Val{T} = Val(Float64), kwargs...) where {T}
+    R_min = T(args[1])
+    R_max = T(args[2])
+    R_v = T(args[3])
+    α = T(args[4])
+    θ = T(2*pi/(5*(n+1)))
+    function f(y)
+        return -T(R_v*pi/n) * sum(y[i] for i = 1:n)
+    end
+    function c(y::V) where {V}
+        ci = zeros(eltype(V), n+2)
+        ci[1] = - R_min*y[1] - y[1]*y[2] + 2*R_min*y[2]*cos(θ)
+        ci[2] = - R_min^2 - R_min*y[1] + 2*R_min*y[1]*cos(θ)
+        ci[3] = - y[n-1]*y[n] - y[n]*R_max + 2*y[n-1]*R_max*cos(θ)
+        ci[4] = - 2*R_max*y[n] + 2*y[n]^2*cos(θ)
```

Comment:
Could you avoid `ci = zeros(eltype(V), n+2)` and use `vcat` to concatenate all the quantities as you did for the other constraint?